pkg/cri/server/snapshots.go

```diff
@@ -49,27 +52,46 @@ func newSnapshotsSyncer(store *snapshotstore.Store, snapshotter snapshot.Snapsho
 	}
 }
 
-// start starts the snapshots syncer. No stop function is needed because
-// the syncer doesn't update any persistent states, it's fine to let it
-// exit with the process.
+// start starts the snapshots syncer.
 func (s *snapshotsSyncer) start() {
-	tick := time.NewTicker(s.syncPeriod)
+	if s.syncTicker != nil {
+		s.syncTicker.Stop()
+	}
+	s.syncTicker = time.NewTicker(s.syncPeriod)
+
+	// build a context for stopping the syncer
+	s.ctx, s.cancel = context.WithCancel(context.TODO())
```

Comment:
`context.TODO()` -> `context.Background()`?

Reply:
Actually, they are same in implementation, I prefer to use `context.TODO()` when the parent context is not clear for sub-jobs. Normally, `context.Backgroud()` is designed for app root context to inherit and should not be used for specific jobs.

Reply:
Agree with Background. TODO looks like we have todo-list here :)